sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -1297,6 +1298,11 @@ def stop_worker(self):
     self.worker_thread.join()
 
 
+# deadlock would happen with subprocess lib, so create a global lock to share
```

Comment:
Perhaps state this more declaratively. E.g. 

# The subprocesses module is not threadsafe on Python 2.7. Use this lock to prevent concurrent calls to POpen.